datashape/promote.py

```diff
@@ -7,7 +7,7 @@
 __all__ = ['promote', 'optionify']
 
 
-def promote(lhs, rhs):
+def promote(lhs, rhs, promote_option=True):
```

Comment:
I think that this is still good behavior in general. I think that in all cases but an inner join, the proper type to promote to would be the option type.